### PR TITLE
src: remove unused headers-exported functions

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
None : generic cleaning
https://redmine.openinfosecfoundation.org/issues/4083

Describe changes:
- remove functions defined in a header and never used (debug, api-completeness, and others)
- remove double definition of IPPairLock

Python helper script to do this in previous PR

#10084 with better fixing of conflicts to get CI green

### CODE REVIEW

File: src/detect-engine-address.c [hunk 1]
```diff
@@ -137,23 +137,6 @@ DetectAddress *DetectAddressCopy(DetectAddress *orig)
...
-void DetectAddressPrintList(DetectAddress *head)
-{
-    SCLogInfo("list:");
-    for (DetectAddress *cur = head; cur != NULL; cur = cur->next) {
-        DetectAddressPrint(cur);
-    }
-    SCLogInfo("endlist");
-}
-#endif
-
```
Comment (jasonish) [inline src/detect-engine-address.c:156]: This one stands out, is it useful when debugging?

**Review by jasonish — APPROVED** [review]
Looks good to me.  Minor concern over some functions that still might be useful when digging into an issue.

File: src/detect-engine-address.c (hunk 1 above)
Comment (catenacyber) [reply src/detect-engine-address.c:156]: It is unused... there is no debug logging calling this. So I would say not so useful..?

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPW2_autofp_stats_chk | | |
| .uptime | 181 | 196 | 108.29% | 

Pipeline 17167

**Comment by pevma** [comment]
Second QA run finished with no deviation.
Pipeline 17172

**Comment by catenacyber** [comment]
Rebased in https://github.com/OISF/suricata/pull/10374

**Pull request closed without merging**